pkg/cloudprovider/controller/nodecontroller.go

```diff
@@ -560,7 +595,7 @@ func (nc *NodeController) MonitorNodeStatus() error {
 				nc.deletePods(node.Name)
 			}
 			if lastReadyCondition.Status == api.ConditionUnknown &&
-				nc.now().After(lastReadyCondition.LastProbeTime.Add(nc.podEvictionTimeout-gracePeriod)) {
+				nc.now().After(nc.nodeStatusMap[node.Name].timestamp.Add(nc.podEvictionTimeout-gracePeriod)) {
```

Comment:
I think we also need to do something about LastTransitionTime (see five lines before this one). It gets written by Kubelet sometimes (search for "LastTransitionTime" in pkg/kubelet/kubelet.go)

One idea would be that the map could be used to store clock deltas rather than absolute times. That is, it's a map of per-node clock skew, and you use it to adjust every absolute time you receive from the Kubelet.